packages/pilot/src/pages/ErrorBoundary/reducer.js

```diff
@@ -0,0 +1,224 @@
+import {
+  __,
+  any,
+  append,
+  concat,
+  defaultTo,
+  difference,
+  either,
+  equals,
+  evolve,
+  head,
+  ifElse,
+  includes,
+  is,
+  map,
+  pipe,
+  prop,
+  props,
+  propSatisfies,
+  reject,
+  test,
+  uncurryN,
+  unless,
+  when,
+} from 'ramda'
+
+import {
+  API_ERROR_RECEIVE,
+  CLEAR_ALL_ERRORS,
+  CLEAR_ERROR,
+  CLEAR_ERRORS_BY_CODE,
+  REACT_ERROR_RECEIVE,
+} from './actions'
+import getResponseError from '../../formatters/apiError'
+import getReactError from '../../formatters/reactError'
+import localErrors from '../../errors'
+
+const initialState = []
+
+const addNewError = uncurryN(2, err => unless(
+  includes(err),
+  append(err)
+))
+
+const removeError = uncurryN(2, err => when(
+  includes(err),
+  reject(equals(err))
+))
+
+const isInCodes = codes => includes(__, codes)
+
+const hasStatusInCodes = codes => propSatisfies(isInCodes(codes), 'status')
+
+const isPrioritaryError = pipe(
+  props(['type', 'priority']),
+  equals(['apiError', true])
+)
+
+const removeErrorsByStatusCode = uncurryN(
+  2,
+  codes => reject(either(
+    hasStatusInCodes(codes),
+    isPrioritaryError
+  ))
+)
+
+const passAnyRoute = uncurryN(2, route => pipe(
+  map(item => item.test(route)),
+  any(equals(true))
+))
+
+const passExactRoute = uncurryN(2, route => pipe(
+  map(pipe(
+    regex => regex.exec(route),
+    defaultTo([]),
+    head,
+    equals(route)
+  )),
+  any(equals(true))
+))
+
+const regexTest = uncurryN(2, propName => pipe(
+  prop(propName),
+  ifElse(
+    is(RegExp),
+    test,
+    equals
+  )
+))
+
+const matchesRegex = uncurryN(2, regex => evolve({
+  message: regexTest('message', regex),
+  method: regexTest('method', regex),
+  name: regexTest('name', regex),
+  status: regexTest('status', regex),
+  type: regexTest('type', regex),
+}))
+
+const matchesApiError = (error, localError) => {
+  const hasRoute = is(Array, localError.affectedRoutes)
+    ? passAnyRoute(error.affectedRoute, localError.affectedRoutes)
+    : true
+
+  const {
+    message,
+    method,
+    name,
+    status,
+    type,
+  } = matchesRegex(localError, error)
+
+  return hasRoute
+    && message
+    && method
+    && name
+    && status
+    && type
+}
+
+const matchesReactError = (error, { affectedRoutes, message, name }) => {
+  const hasRoute = is(Array, affectedRoutes)
+    ? passAnyRoute(error.affectedRoute, affectedRoutes)
+    : true
+
+  const matchesMessage = is(RegExp, message) && message.test(error.message)
+  const matchesName = is(RegExp, name) && name.test(error.name)
+
+  return hasRoute && matchesMessage && matchesName
+}
+
+const findError = (error, errors) => {
+  const sameRouteErrors = errors.filter(({ affectedRoutes }) =>
+    passExactRoute(error.affectedRoute, affectedRoutes))
+
+  const differentRouteErrors = difference(errors, sameRouteErrors)
+  const sameRouteErrorsFirst = concat(sameRouteErrors, differentRouteErrors)
+
+  const foundError = sameRouteErrorsFirst.find((localError) => {
```

Comment:
We can assume that if the error isn't an API error it's automatically a React error because we are working only with these 2 types.
```suggestion
  const foundError = sameRouteErrorsFirst.find((localError) => {
     if (error.source === 'api') {
      return matchesApiError(error, localError)
    } 
      return matchesReactError(error, localError)
  })
```